Compatibility with vtk 9.1 and higher

I tried compiling Elmer using VTK v9.3.1 in Arch Linux but ran into trouble.  After a quick search, I found that the file `vtkConfigure.h` had been [deprecated since v9.1](https://gitlab.kitware.com/vtk/vtk/-/blob/master/Documentation/release/9.1.md#filters-2).  I also found that this issue was spotted earlier in #308, but the modifications were not merged due to compatibility issues with older versions of VTK. 
Based on the PR, I made this patch that should be backward compatible:

```
diff --unified --recursive --text a/ElmerGUI/Application/cad/cadview.h b/ElmerGUI/Application/cad/cadview.h
--- a/ElmerGUI/Application/cad/cadview.h	2024-08-20 14:30:05.855951677 -0600
+++ b/ElmerGUI/Application/cad/cadview.h	2024-08-21 13:48:40.130823655 -0600
@@ -58,11 +58,14 @@
 #include <BRepMesh_IncrementalMesh.hxx>
 #endif
 
-#include "vtkConfigure.h"
 #ifndef VTK_MAJOR_VERSION
 #include "vtkVersionMacros.h"
 #endif
 
+#if VTK_MAJOR_VERSION <= 9 && VTK_MINOR_VERSION < 1
+#include "vtkConfigure.h"
+#endif
+
 class QMenu;
 class QAction;
 #if VTK_MAJOR_VERSION >= 8
diff --unified --recursive --text a/ElmerGUI/Application/vtkpost/vtkpost.h b/ElmerGUI/Application/vtkpost/vtkpost.h
--- a/ElmerGUI/Application/vtkpost/vtkpost.h	2024-08-20 14:30:05.875951676 -0600
+++ b/ElmerGUI/Application/vtkpost/vtkpost.h	2024-08-21 13:48:59.070823122 -0600
@@ -57,11 +57,14 @@
 #include <gui/PythonQtScriptingConsole.h>
 #endif
 
-#include "vtkConfigure.h"
 #ifndef VTK_MAJOR_VERSION
 #include "vtkVersionMacros.h"
 #endif
 
+#if VTK_MAJOR_VERSION <= 9 && VTK_MINOR_VERSION < 1
+#include "vtkConfigure.h"
+#endif
+
 class EpMesh;
 class ScalarField;
 #if VTK_MAJOR_VERSION >= 8
```
So far, so good. However, now I have a different error regarding `matc`:
```
/usr/bin/ld: listheaders: TLS definition in matc/src/libmatc.so section .tbss mismatches non-TLS reference in /tmp/ccsoZVGb.ltrans30.ltrans.o
/usr/bin/ld: /tmp/ccsoZVGb.ltrans30.ltrans.o: error adding symbols: bad value
collect2: error: ld returned 1 exit status
```
By the way, this error is not thrown when the compilation is made without VTK. I am currently trying to pinpoint the problem, but I haven't had any luck.

Any insights?

Cheers

--- a/.github/workflows/build-windows-mingw.yaml
+++ b/.github/workflows/build-windows-mingw.yaml
@@ -61,10 +61,27 @@ jobs:
             fc:p
             cmake:p
             msmpi:p
+            opencascade:p
             openblas:p
             parmetis:p
             qt6-declarative:p
             qwt-qt6:p
+            vtk:p
+            adios2:p
+            boost:p
+            cgns:p
+            cli11:p
+            eigen3:p
+            fast_float:p
+            ffmpeg:p
+            gl2ps:p
+            liblas:p
+            libmariadbclient:p
+            openslide:p
+            openvdb:p
+            pdal:p
+            unixodbc:p
+            utf8cpp:p
             ${{ matrix.umfpack-package }}
 
       - name: install MSMPI
@@ -95,6 +112,7 @@ jobs:
             -DWITH_Mumps=OFF \
             -DWITH_ELMERGUI=ON \
             -DWITH_QT6=ON \
+            -DWITH_VTK=ON \
             -DWITH_PARAVIEW=ON \
             -DCREATE_PKGCONFIG_FILE=ON \
             ${{ matrix.umfpack-cmake-flags }} \

--- a/.github/workflows/build-windows-mingw.yaml
+++ b/.github/workflows/build-windows-mingw.yaml
@@ -61,7 +61,6 @@ jobs:
             fc:p
             cmake:p
             msmpi:p
-            opencascade:p
             openblas:p
             parmetis:p
             qt6-declarative:p
@@ -82,6 +81,7 @@ jobs:
             pdal:p
             unixodbc:p
             utf8cpp:p
+            opencascade:p
             ${{ matrix.umfpack-package }}
 
       - name: install MSMPI
@@ -113,6 +113,7 @@ jobs:
             -DWITH_ELMERGUI=ON \
             -DWITH_QT6=ON \
             -DWITH_VTK=ON \
+            -DWITH_OCC=ON \
             -DWITH_PARAVIEW=ON \
             -DCREATE_PKGCONFIG_FILE=ON \
             ${{ matrix.umfpack-cmake-flags }} \

--- a/.github/workflows/build-windows-mingw.yaml
+++ b/.github/workflows/build-windows-mingw.yaml
@@ -63,8 +63,8 @@ jobs:
             msmpi:p
             openblas:p
             parmetis:p
-            qwt-qt5:p
-            qt5-script:p
+            qt6-declarative:p
+            qwt-qt6:p
             ${{ matrix.umfpack-package }}
 
       - name: install MSMPI
@@ -94,6 +94,7 @@ jobs:
             -DParMetis_INCLUDE_DIR="$(pkg-config --cflags parmetis)" \
             -DWITH_Mumps=OFF \
             -DWITH_ELMERGUI=ON \
+            -DWITH_QT6=ON \
             -DWITH_PARAVIEW=ON \
             -DCREATE_PKGCONFIG_FILE=ON \
             ${{ matrix.umfpack-cmake-flags }} \

--- a/ElmerGUI/Application/cad/cadview.h
+++ b/ElmerGUI/Application/cad/cadview.h
@@ -58,7 +58,6 @@ namespace nglib {
 #include <BRepMesh_IncrementalMesh.hxx>
 #endif
 
-#include "vtkConfigure.h"
 #ifndef VTK_MAJOR_VERSION
 #include "vtkVersionMacros.h"
 #endif

--- a/ElmerGUI/Application/vtkpost/vtkpost.h
+++ b/ElmerGUI/Application/vtkpost/vtkpost.h
@@ -57,7 +57,6 @@
 #include <gui/PythonQtScriptingConsole.h>
 #endif
 
-#include "vtkConfigure.h"
 #ifndef VTK_MAJOR_VERSION
 #include "vtkVersionMacros.h"
 #endif

